core/src/apps/management/backup_device.py

```diff
@@ -18,14 +18,19 @@ async def backup_device(msg: BackupDevice) -> Success:
     if not storage_device.needs_backup():
         raise wire.ProcessError("Seed already backed up")
 
-    mnemonic_secret, mnemonic_type = mnemonic.get()
+    mnemonic_secret, backup_type = mnemonic.get()
     if mnemonic_secret is None:
         raise RuntimeError
 
     storage_device.set_unfinished_backup(True)
     storage_device.set_backed_up()
 
-    await backup_seed(mnemonic_type, mnemonic_secret)
+    await backup_seed(
```

Comment:
...now you should be checking, before you call `backup_seed`, the validity of arguments
(i.e., BIP39 has no groups and no group threshold; if group threshold is specified, at least that many groups are specified, etc.)

Reply:
Why here and not *inside* `backup_seed`: https://github.com/trezor/trezor-firmware/pull/3710/files#diff-35a83e5f83f9aef28cfd412567ff6f0ae1fd985bba5779cc682f0f38fb6d47ffR223

Reply:
Separation of responsibilities? `backup_seed` just asserts that its arguments are correct and then does its thing -- which is, specifically, directing the parameters to the appropriate layout function. Adding the validation in there is (a) pushing _user-facing errors_ further away from the user-provided values, and (b) tacking on new functionality onto `backup_seed`, namely, caring for user-generated error states.

Speaking of, maybe `backup_seed` should stay unmodified and we should do "if parameters are provided then launch `_backup_slip39` directly" inline here. (the other code path for `backup_seed` doesn't care about that functionality)

Reply:
39e10c48c0c2978f199d4847a423bf14b36fa8e3